### PR TITLE
Panels

### PR DESCRIPTION
**Pull request by bunsenstraat**

### CODE REVIEW

File: apps/remix-ide/src/app.js [hunk 1]
```diff
@@ -182,22 +221,28 @@ class AppComponent {
 
     // LAYOUT & SYSTEM VIEWS
     const appPanel = new MainPanel()
-    self.mainview = new MainView(contextualListener, editor, appPanel, fileManager, appManager, terminal)
+    // self.mainview = new MainView(contextualListener, editor, appPanel, fileManager, appManager, terminal)
     Registry.getInstance().put({ api: self.mainview, name: 'mainview' })
```
Comment (yann300) [inline apps/remix-ide/src/app.js:225]: This can be removed too I think

File: apps/remix-ide/src/app.js [hunk 2]
```diff
@@ -263,61 +317,86 @@ class AppComponent {
...
+      'theme',
+      'fileManager',
+      'compilerMetadata',
+      'compilerArtefacts',
+      'network',
+      'web3Provider',
+      'offsetToLineColumnConverter'
+    ])
+    await self.appManager.activatePlugin(['mainPanel'])
     await self.appManager.activatePlugin(['mainPanel', 'menuicons', 'tabs'])
```
Comment (yann300) [inline apps/remix-ide/src/app.js:334]: it's already being activated above

File: apps/remix-ide/src/app/components/side-panel.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    this.emit('focusChanged', name)
+    this.renderComponent()
+  }
+
+  render () {
+    return this.sideelement
+  }
+
+  renderComponent () {
+    ReactDOM.render(<RemixPluginPanel header={<RemixUIPanelHeader plugins={this.plugins}></RemixUIPanelHeader>} plugins={this.plugins}/>, this.sideelement)
```
Comment (yann300) [inline apps/remix-ide/src/app/components/side-panel.tsx:93]: I don't think `</RemixUIPanelHeader>` is needed here

Comment (yann300) [reply apps/remix-ide/src/app/components/side-panel.tsx:93]: or wait yes my bad, this is fine

File: libs/remix-ui/terminal/src/lib/remix-ui-terminal.tsx [hunk 4]
```diff
@@ -105,9 +87,15 @@ export const RemixUiTerminal = (props: RemixUiTerminalProps) => {
     props.onReady({
       logHtml: (html) => {
         scriptRunnerDispatch({ type: 'html', payload: { message: [html.innerText] } })
+
+        setTimeout(() => {
```
Comment (yann300) [inline libs/remix-ui/terminal/src/lib/remix-ui-terminal.tsx:91]: did you want to add `scriptRunnerDispatch` inside it?

**Comment by yann300** [comment]
it seems the url parameter `embed` and `minimizesidepanel` sill need to be added.

**Comment by bunsenstraat** [comment]
@yann300 what does embed parameter do?

**Comment by bunsenstraat** [comment]
@yann300 minimizesidepanel is not covered by any test, neither is minimizeterminal

**Comment by yann300** [comment]
embed, just minimize terminal and side panel.

File: libs/remix-ui/panel/src/lib/plugins/panel.css [hunk 5]
```diff
@@ -0,0 +1,110 @@
+.panel {
```
Comment (LianaHus) [inline libs/remix-ui/panel/src/lib/plugins/panel.css:1]: most of those properties you can set as a bootstrap class, which is preferable

File: libs/remix-ui/panel/.eslintrc.json [hunk 6]
```diff
@@ -0,0 +1,249 @@
+{
```
Comment (ioedeveloper) [inline libs/remix-ui/panel/.eslintrc.json:1]: Update linting rules to default.

```
{
  "extends": ["plugin:@nrwl/nx/react", "../../../.eslintrc"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
```

**Review by ioedeveloper — CHANGES_REQUESTED** [review]

**Review by LianaHus — APPROVED** [review]

**Review by ioedeveloper — APPROVED** [review]

**Pull request merged**